### PR TITLE
CHAOS-328: Add ddmark to the validation webhook

### PR DESCRIPTION
**Pull request by nathantournant**
## What does this PR do?

- [ ] Adds new functionality
- [X] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

### Please briefly describe your changes as well as the motivation behind them:
`ddmark` could not be added to the validation webhook because it uses the go installation of the machine it runs on to process the libraries it reads the markers on.

Changes in this PR:
 - `ddmark`: 
   - still needs to run on a `go` environment
   - `ddmark` now reads libraries from an `Embed.FS` input. It's a minimal change on the analyzed libraries - which code includes ddmark-specific markers anyway.
   - libraries need to be imported as `Embed.FS` variables through the `ddmark.InitLibrary()` function and discarded through `ddmark.CleanupLibraries()`.
 - `chaosli`:
   - `chaosli` adapted to the change of ddmark, losing the logic of copying files in the gopath of the machine it runs on.
 - `chaos-controller`:
   - `ddmark` validation is now included in the validation webhook as it should've been for long.
   - base image changed from `distroless` to `go-alpine:1.18`, smallest go image I know of.

Unit tests were adapted because of the ddmark use changes.
E2E tests were adapted because of the controller ddmark rules enforcement.

## Code Quality Checklist

- [X] The documentation is up to date.
- [X] My code is sufficiently commented and passes continuous integration checks.
- [X] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [X] I leveraged continuous integration testing
    - [X] by depending on existing `unit` tests or `end-to-end` tests.
    - [X] by adding new `unit` tests or `end-to-end` tests.
- [X] I manually tested the following steps:
    - Attempt to apply disruptions triggering specific `ddmark` validations eg. a combined NodeFailure and CPUPressure disruption.
    - [X] locally.

### CODE REVIEW

File: controllers/disruption_controller_test.go [hunk 1]
```diff
@@ -168,12 +168,6 @@ var _ = Describe("Disruption Controller", func() {
 				Selector:   map[string]string{"foo": "bar"},
 				Containers: []string{"ctn1"},
 				Duration:   "10m",
-				NodeFailure: &chaosv1beta1.NodeFailureSpec{
```
Comment (ptnapoleon) [inline controllers/disruption_controller_test.go:171]: Why did these need to be removed?

Comment (nathantournant) [reply controllers/disruption_controller_test.go:171]: this disruption is blocked by the ddmark validation -- as it should be, it's not allowed by the rules we set up
should I change the e2e test structure to try and get better e2e coverage following this update ?

File: api/v1beta1/disruption_webhook.go [hunk 2]
```diff
@@ -39,8 +40,13 @@ var namespaceThreshold float64
 var clusterThreshold float64
 var handlerEnabled bool
 var defaultDuration time.Duration
+var ddmarkLibPath = "chaos-api"
```
Comment (ptnapoleon) [inline api/v1beta1/disruption_webhook.go:43]: Any chance of making this a const and defining it in only one place across the codebase?

File: ddmark/libtools.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
```
Comment (ptnapoleon) [inline ddmark/libtools.go:4]:
```suggestion
// Copyright 2022 Datadog, Inc.
```

File: api/v1beta1/disruption_webhook.go [hunk 4]
```diff
@@ -102,6 +108,11 @@ func (r *Disruption) ValidateCreate() error {
 		return err
 	}
 
+	multiErr := ddmark.ValidateStructMultierror(r.Spec, "test_suite", ddmarkLibPath)
```
[the remark this answers is not in the document]
Comment (nathantournant) [reply api/v1beta1/disruption_webhook.go:111]: you're right: i should put `validation_webhook` there
it's an arbitrary value anyway, that will show the location in the ddmark logs

File: api/v1beta1/disruption_webhook.go (hunk 2 above)
Comment (nathantournant) [reply api/v1beta1/disruption_webhook.go]: added to types, it is easier this way

File: ddmark/libtools.go (hunk 3 above)
Comment (nathantournant) [reply ddmark/libtools.go:4]: Separate PR

**Review by ptnapoleon — APPROVED** [review]

**Pull request merged**